aten/src/ATen/core/type.cpp

```diff
@@ -141,51 +141,39 @@ ListTypePtr ListType::ofBools() {
   return value;
 }
 
-c10::optional<TypePtr> tryEitherIsTheSuperType(const TypePtr& t1, const TypePtr& t2) {
-  if (t1->isSubtypeOf(t2)) {
-    return t2;
-  } else if (t2->isSubtypeOf(t1)) {
-    return t1;
-  } else {
-    return c10::nullopt;
-  }
-}
 
-c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2) {
-  //cases that t1 == t2, or t1 is a type refinement of t2 and vice versa
-  if (auto maybe_supertype = tryEitherIsTheSuperType(t1, t2)) {
-    return *maybe_supertype;
-  }
-
-  // NB: we do not return NumberType because there is not currently enough
-  // operator support for it
+c10::optional<TypePtr> unifyTypes(const TypePtr& type1_input, const TypePtr& type2_input) {
+  auto t1 = type1_input;
+  auto t2 = type2_input;
 
+  // Handle non-container types which do not subtype each other and unify
   if (t1->kind() == TensorType::Kind && t2->kind() == TensorType::Kind) {
-    return t1->expect<TensorType>()->merge(t2->expect<TensorType>());
+    return type1_input->expect<TensorType>()->merge(type2_input->expect<TensorType>());
```

Comment:
I'd make the test and merge consistently use either type1/2_input or t1/t2, presumably the latter. (But if it's important to use the former, I'd move the declaration of t1/t2 down below just for clarity.)